### ISSUE TITLE
Setup JHM profiler

### ISSUE THREAD

**Issue by mcpiroman** [Kotlin/kotlinx-benchmark#50]
For Jvm-only setup, can I use profiler as described in https://github.com/openjdk/jmh/blob/master/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_35_Profilers.java?

**Comment by elect86** [comment]
Any news on this?

**Comment by max-kammerer** [comment]
@mcpiroman @elect86 Could you clarify you question?
There is simple example with JMH https://github.com/Kotlin/kotlinx-benchmark/blob/master/examples/java/src/main/java/test/SampleJavaBenchmark.java, maybe it could answer your question?

**Comment by elect86** [comment]
Nope, [this](https://github.com/openjdk/jmh/blob/master/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_35_Profilers.java#L125-L126) is what we'd like

**Comment by max-kammerer** [comment]
@elect86 Sorry, but pointing to line doesn't explain your problem and difficulties. Could you explain them?

**Comment by elect86** [comment]
Ok, I'll try to explain better.

From the main docs in the sample

>  JMH has a few very handy profilers that help to understand your benchmarks. While
    these profilers are not the substitute for full-fledged external profilers, in many
    cases, these are handy to quickly dig into the benchmark behavior. When you are
    doing many cycles of tuning up the benchmark code itself, it is important to have
    a quick turnaround for the results.
    
Among these, there is for example one regarding the GC. You can use it by passing `-prof gc` on the cli.
It'll something like [here](https://shipilev.net/jvm/anatomy-quarks/18-scalar-replacement/):

> Benchmark                                      Mode  Cnt     Score    Error   Units
>
> ScalarReplacement.single                       avgt   15     1.919 ±  0.002   ns/op
> ScalarReplacement.single:·gc.alloc.rate        avgt   15    ≈ 10⁻⁴           MB/sec
> ScalarReplacement.single:·gc.alloc.rate.norm   avgt   15    ≈ 10⁻⁶             B/op
> ScalarReplacement.single:·gc.count             avgt   15       ≈ 0           counts

It's useful to check, for example, if the EA kicks in and helps avoiding allocations

I'd like to have the possibility to set it in kotlinx-benchmark as well. I tried `param` and `advanced`, but it didn't work

**Comment by max-kammerer** [comment]
@elect86 Thank you for clarification! I hope @qurbonzoda could answer your question after coming back from vacation

**Comment by guohao** [comment]
Profiler is a very important feature for JMH users,  hope to support soon :)

**Comment by qurbonzoda** [comment]
Hi everyone,
You can still run a specific profiler (`-prof <profiler>`) by running the jar file that the `<target>BenchmarkJar` task generates. The generated jar file is located under the `build/benchmarks/<target>/jars` directory.

Currently, there is no way to define a profiler in the configuration script. Would the configuration option be helpful for you?
